src/pipecat/frames/frames.py

```diff
@@ -1289,6 +1266,100 @@ class SpeechControlParamsFrame(SystemFrame):
     turn_params: Optional[SmartTurnParams] = None
 
 
+#
+# Task frames
+#
+
+
+@dataclass
+class TaskFrame(SystemFrame):
+    """Base frame for task frames.
+
+    This is a base class for frames that are meant to be sent and handled
```

Comment:
If we _do_ eventually want to have some generic mechanisms built around `TaskFrame`s (like `await self.push_task_frame_and_wait(InterruptionTaskFrame())`), is this too loose a definition for task frames?

Would it be useful to define it more narrowly, like:

```python
This is a base class for frames that are meant to be sent upstream as requests to the pipeline task to trigger equivalent downstream frames (e.g. `InterruptionTaskFrame` == "please send an `InterruptionFrame` downstream`)
```